[RFE] Automate Services backwards compatibility change for dialog options.

tinaafitz opened the pull request:
Add call to instantiate dialog options if parsed dialog options are not found.

https://bugzilla.redhat.com/show_bug.cgi?id=1216210

gmcculloug commented on vmdb/db/fixtures/ae_datastore/ManageIQ/Service/Provisioning/StateMachines/Methods.class/__methods__/catalogbundleinitialization.rb:
```diff
@@ -68,14 +68,23 @@ def tag_service(dialogs_tags_hash)
+
+def parsed_dialog_information
+  dialog_options_hash = @task.get_option(:parsed_dialog_options).nil? ? nil : yaml_load(:parsed_dialog_options)
+  dialog_tags_hash = @task.get_option(:parsed_dialog_tags).nil? ? nil : yaml_load(:parsed_dialog_tags)
```
You meant to call the `yaml_data` method on these two lines.  Also, since the yaml_data method is designed to get the data from the task's options hash I would move the nil check logic into that method as well.

gmcculloug commented on vmdb/db/fixtures/ae_datastore/ManageIQ/Service/Provisioning/StateMachines/Methods.class/__methods__/catalogiteminitialization.rb:
```diff
@@ -193,14 +193,25 @@ def merge_dialog_information(dialog_options_hash, dialog_tags_hash)
+
+def parsed_dialog_information
+  dialog_options_hash = @task.get_option(:parsed_dialog_options).nil? ? nil : yaml_data(:parsed_dialog_options)
+  dialog_tags_hash = @task.get_option(:parsed_dialog_tags).nil? ? nil : yaml_data(:parsed_dialog_tags)
```
Same here.  Move the nil check logic into the yaml_data method.

tinaafitz commented:
@gmcculloug
 Corrected an issue I found while testing changes per your comments.  Changed the call from parsed_dialog_information to yaml_data after instantiating DialogParser.
